pkg/daemon/ceph/config/config.go

```diff
@@ -98,6 +98,15 @@ func GenerateAdminConnectionConfigWithSettings(context *clusterd.Context, cluste
 		return "", errors.Wrapf(err, "failed to write keyring to %s", root)
 	}
 
+	// If this is an external cluster
+	if cluster.IsInitializedExternalCred() {
```

Comment:
I wonder if we can treat the external cluster the same as a normal cluster. The operator has a user/key pair that it uses for all operations. For a normal cluster it will always be client.admin. For the external cluster it can be something else. But it seems like for both cases it should be the same code path to write the keyring instead of having a special case for an external cluster.

Reply:
As discussed we will do this in master.